src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import com.example.onlinebookstore.mapper.CategoryMapper;
+import com.example.onlinebookstore.model.Category;
+import com.example.onlinebookstore.repository.CategoryRepository;
+import com.example.onlinebookstore.service.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return categoryMapper.toDto(categoryRepository.findById(id).orElseThrow(() ->
+                new RuntimeException("There is not any category in db by id %d"
```

Comment:
maybe, you should use a custom exception?

Reply:
agree. Let’s  fix it